tsconfig.json

```diff
@@ -1,39 +1,34 @@
 {
   "compilerOptions": {
     "allowJs": true,
-    "allowSyntheticDefaultImports": true,
-    "baseUrl": ".",
-    "esModuleInterop": true,
-    "inlineSources": true,
+    // ensures single-file compilers, like babel, webpack+swc, etc work
     "isolatedModules": true,
+    // makes typechecking faster (also see `tsBuildInfoFile` setting)
+    "incremental": true,
     "jsx": "react",
-    "lib": ["dom", "es2020"],
-    "moduleResolution": "node",
+    // todo: make two tsconfigs, one for node (development and test) and one for
+    // browser (app, ui, shared?) and ensure the code is split up into the right
+    // directories that make sense for their environment.
+    // I think this would eliminate a lot of eslint overrides complexity
+    // In the meantime, we need DOM libs and should target the ES version in
+    // "@tsconfig/node20/tsconfig.json" (`libs` here overwrites extends's `libs`)
+    "lib": ["DOM", "es2023"],
```

Comment:
I've added a TODO here because we should probably have two tsconfigs one day. When we are writing Node.js code we should have DOM globals available.

Reply:
Yes. I've fixed my comment. Thank you!